cobalt/watchdog/watchdog.cc

```diff
@@ -541,35 +544,32 @@ bool Watchdog::Ping(const std::string& name, const std::string& info) {
 
 std::string Watchdog::GetWatchdogViolations(
     const std::vector<std::string>& clients, bool clear) {
-  // Gets a json string containing the Watchdog violations since the last
-  // call (up to the kWatchdogMaxViolations limit).
+  // Gets a json string containing the Watchdog violations of the given clients
+  // since the last call (up to the kWatchdogMaxViolations limit).
   if (is_disabled_) return "";
 
-  std::string watchdog_json = "";
-  std::string watchdog_json_fetched = "";
+  std::string fetched_violations_json = "";
 
   starboard::ScopedLock scoped_lock(mutex_);
 
-  if (pending_write_) WriteWatchdogViolations();
-
   if (!static_cast<base::DictionaryValue*>(violations_map_.get())->empty()) {
-    // Get all Watchdog violations if clients is given.
+    // Gets all Watchdog violations if no clients are given.
     if (clients.empty()) {
-      // Removes all Watchdog violations.
-      base::JSONWriter::Write(*violations_map_, &watchdog_json_fetched);
+      base::JSONWriter::Write(*violations_map_, &fetched_violations_json);
       if (clear) {
         static_cast<base::DictionaryValue*>(violations_map_.get())->Clear();
         violations_count_ = 0;
         starboard::SbFileDeleteRecursive(GetWatchdogFilePath().c_str(), true);
       }
+      // Gets all Watchdog violations of the given clients.
```

Comment:
If clear is false, we should also call EvictOldWatchdogViolations()?

Reply:
agreed, if clear is false we shouldn't touch the json file (including evicting old violations).